.github/workflows/run-and-track-benchmarks-on-main.yaml

```
@@ -5,69 +5,68 @@ on:
     branches:
       - main
 
+permissions:
+  checks: write
+
 jobs:
   benchmark_sv1_criterion_with_bencher:
     name: Track sv1 criterion benchmarks with Bencher
     runs-on: ubuntu-latest
     env:
-        BENCHER_PROJECT: stratum-v2-sri
-        BENCHER_API_TOKEN: ${{ secrets.BENCHER_API_TOKEN }}
-        BENCHER_ADAPTER: rust_criterion
-        BENCHER_TESTBED: sv1
+      BENCHER_PROJECT: stratum-v2-sri
+      BENCHER_API_TOKEN: ${{ secrets.BENCHER_API_TOKEN }}
+      BENCHER_ADAPTER: rust_criterion
+      BENCHER_TESTBED: sv1
     steps:
       - name: Set up Rust
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: 1.75.0
           override: true
-      - name: Checkout repository
-        uses: actions/checkout@v2
-            
+
       - uses: actions/checkout@v4
```

```suggestion
      - name: Checkout repository
        uses: actions/checkout@v4
```
Keeping it this way would maintain consistency with other actions.